src/main/java/securbank/controller/CommonController.java

```diff
@@ -1,86 +1,121 @@
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import securbank.models.ChangePasswordRequest;
+import securbank.models.User;
+import securbank.services.UserService;
+import securbank.validators.NewUserFormValidator;
+/**
+ * @author Ayush Gupta
+ *
+ */
+@Controller
+public class CommonController {
+	@Autowired
+	UserService userService;
+	
+	@Autowired 
+	NewUserFormValidator userFormValidator;
+	
+	final static Logger logger = LoggerFactory.getLogger(CommonController.class);
+	
+	
+	@RequestMapping(value="/logout", method = RequestMethod.GET)
+	public String logoutPage (HttpServletRequest request, HttpServletResponse response) {
+	    Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+	    if (auth != null){    
+	        new SecurityContextLogoutHandler().logout(request, response, auth);
+	    }
+	    return "redirect:/login?logout";//You can redirect wherever you want, but generally it's a good practice to show login screen again.
+	}
+	
+	@GetMapping("/signup")
+    public String signupForm(Model model) {
+		model.addAttribute("user", new User());
+		logger.info("GET request: signup form");
+		
+		return "signup";
+    }
+		
+	@PostMapping("/signup")
+    public String signupSubmit(@ModelAttribute User user, BindingResult bindingResult) {
+		userFormValidator.validate(user, bindingResult);
+		if (bindingResult.hasErrors()) {
+			logger.info("POST request: signup form with validation errors");
+			
+			return "signup";
+        }
+		
+		logger.info("POST request: signup");
+		logger.info("Username: "+user.getUsername());
+		
+    	userService.createExternalUser(user);
+    	
+        return "redirect:/";
+    }
+	
+	@GetMapping("/user/verify/{id}")
+    public String verifyNewUser(Model model, @PathVariable UUID id) {
+		if (userService.verifyNewUser(id) == false) {
+			logger.info("GET request: verification failed of new external user");
+			return "redirect:/error?code=400";
+		}
+		logger.info("GET request: verification of new external user");
+		
+		return "redirect:/";
+    }
+	
+	@GetMapping("/changepassword")
+	public String ChangePasswordform(Model model){
+		User user = userService.getCurrentUser();
+		if (user == null) {
+			logger.info("GET request: verification failed for user");
+			return "redirect:/error?code=400&path=user.notfound";
+		}		
+		model.addAttribute("changePasswordRequest", new ChangePasswordRequest());
+		logger.info("GET request: Change password");
+			
+        return "changepassword";
+		
+	}
+	
+	@PostMapping("/changepassword")
+    public String changeUserPassword(@ModelAttribute ChangePasswordRequest request, BindingResult binding) {		
+		if(request.getNewPassword().compareTo(request.getConfirmPassword()) == 0){		

```

Comment:
Create a validator to confirm that passwords are in the format defined by 
https://github.com/ayush2k/securbank/blob/master/src/main/java/securbank/utils/ContraintUtils.java#L60